starter-code/pizza.js

```diff
@@ -45,29 +45,128 @@ function renderPepperonni() {
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function($mushroom) {
+    if (state.mushrooms) {
+      $mushroom.style.visibility = "visible";
+    }
+    else {
+      $mushroom.style.visibility = "hidden";
+    }
+  })
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(function($greenPepper) {
+    if (state.greenPeppers) {
+      $greenPepper.style.visibility = "visible";
+    }
+    else {
+      $greenPepper.style.visibility = "hidden";
+    }
+  })
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  if (state.whiteSauce) {
+    document.querySelector('.sauce').classList.add('sauce-white')
+  }
+  else {
+    document.querySelector('.sauce').classList.remove('sauce-white')
+  }
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  if (state.glutenFreeCrust) {
+    document.querySelector('.crust').classList.add('crust-gluten-free')
+  }
+  else {
+    document.querySelector('.crust').classList.remove('crust-gluten-free')
+  }
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  if (state.pepperonni){
+    document.querySelector('.btn.btn-pepperonni').classList.add('active')
+  }
+  else {
+    document.querySelector('.btn.btn-pepperonni').classList.remove('active')
+  }
+  if (state.mushrooms){
+    document.querySelector('.btn.btn-mushrooms').classList.add('active')
+  }
+  else {
+    document.querySelector('.btn.btn-mushrooms').classList.remove('active')
+  }
+  if (state.greenPeppers){
+    document.querySelector('.btn.btn-green-peppers').classList.add('active')
+  }
+  else {
+    document.querySelector('.btn.btn-green-peppers').classList.remove('active')
+  }
+  if (state.whiteSauce){
+    document.querySelector('.btn.btn-sauce').classList.add('active')
+  }
+  else {
+    document.querySelector('.btn.btn-sauce').classList.remove('active')
+  }
+  if (state.glutenFreeCrust){
+    document.querySelector('.btn.btn-crust').classList.add('active')
+  }
+  else {
+    document.querySelector('.btn.btn-crust').classList.remove('active')
+  }
 }
 
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
+  let totalPrice = 10;
+
```

Comment:
see comment above :)

Reply:
Also you added classes to your li tags, which works but is maybe not the best practice. I would suggest you would loop over the state and depending on the state of the ingredient create a new li tag and set the innerHTML to your wanted text using the ingredients object given to you at the top of your js file. (Same as you did for calculating the price)